### PR TITLE
[Merged by Bors] - chore (Data.Fintype.Basic): de-simp `Fintype.univ_ofSubsingleton`

### PR DESCRIPTION
**Pull request by mattrobball**
In principle, `Fintype.univ_ofSubsingleton` could be a `simp` theorem but it applies to any occurence of `univ` and requires unification of the (possibly very complex) `Fintype` instances. It appears to not be needed in `simp`. So we remove the attribute. We get some performance gains as a benefit. 

---

See the [Zulip topic](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60simp.60.20vs.20subsingleton.20instances) for an example of it causing problems in the wild.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/CategoryTheory/Preadditive/Mat.lean [hunk 1]
```diff
@@ -341,7 +341,7 @@ def isoBiproductEmbedding (M : Mat_ C) : M ≅ ⨁ fun i => (embedding C).obj (M
     rw [Finset.sum_apply, Finset.sum_apply, Finset.sum_eq_single i]; rotate_left
     · intro b _ hb
       dsimp
-      simp only [Finset.sum_singleton]
+      erw [Finset.sum_singleton]
```
Comment (b-mehta) [inline Mathlib/CategoryTheory/Preadditive/Mat.lean:344]: Can we use `rw [Finset.univ_ofSubsingleton, Finset.sum_singleton]` here? I'd like to reduce usage `erw` if possible.

Comment (mattrobball) [reply Mathlib/CategoryTheory/Preadditive/Mat.lean]: Yes indeed! I tried turning it on as an instance here so assumed something else was going on. But it is good. Thanks.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**